index.js

```diff
@@ -1016,41 +1020,63 @@ const createFocusTrap = function (elements, userOptions) {
       return this;
     },
 
+    _isManuallyPaused() {
+      return state.manuallyPaused;
+    },
+
+    _setPausedState(paused, options) {
+      state.paused = paused;
+      if (paused) {
+        const onPause = getOption(options, 'onPause');
+        const onPostPause = getOption(options, 'onPostPause');
+        onPause?.();
+
+        removeListeners();
+        updateObservedNodes();
+
+        onPostPause?.();
+      } else {
+        const onUnpause = getOption(options, 'onUnpause');
+        const onPostUnpause = getOption(options, 'onPostUnpause');
+
+        onUnpause?.();
+
+        updateTabbableNodes();
+        addListeners();
+        updateObservedNodes();
+
+        onPostUnpause?.();
+      }
+
+      return this;
+    },
+
     pause(pauseOptions) {
-      if (state.paused || !state.active) {
+      if (!state.active) {
         return this;
       }
 
-      const onPause = getOption(pauseOptions, 'onPause');
-      const onPostPause = getOption(pauseOptions, 'onPostPause');
-
-      state.paused = true;
-      onPause?.();
+      state.manuallyPaused = true;
 
-      removeListeners();
-      updateObservedNodes();
+      if (state.paused) {
+        return this;
+      }
 
-      onPostPause?.();
-      return this;
+      return this._setPausedState(true, pauseOptions);
     },
 
     unpause(unpauseOptions) {
-      if (!state.paused || !state.active) {
+      if (!state.active) {
         return this;
       }
 
-      const onUnpause = getOption(unpauseOptions, 'onUnpause');
-      const onPostUnpause = getOption(unpauseOptions, 'onPostUnpause');
-
-      state.paused = false;
-      onUnpause?.();
+      state.manuallyPaused = false;
 
-      updateTabbableNodes();
-      addListeners();
-      updateObservedNodes();
+      if (!state.paused || trapStack[trapStack.length - 1] !== this) {
```

Comment:
Can simplify to just `if (trapStack[trapStack.length - 1] !== this)` here with the state check added into `_setPausedState()`.

Reply:
Done